### ISSUE TITLE
`latency jump` logs spamming the cockroach.log

### ISSUE THREAD

**Issue by kevinkokomani** [cockroachdb/cockroach#93999]
**Describe the problem**

I received a debug zip and the log files were filled to the brim with `latency jump` log files:

```
847921 latency jump (prev avg 11.55ms, current 96.81ms)
847922 latency jump (prev avg 3.36ms, current 7.86ms)
847923 latency jump (prev avg 5.25ms, current 52.26ms)
847924 latency jump (prev avg 6.45ms, current 11.38ms)
847925 latency jump (prev avg 4.19ms, current 24.40ms)
847926 latency jump (prev avg 4.94ms, current 10.36ms)
847927 latency jump (prev avg 7.01ms, current 18.65ms)
847928 latency jump (prev avg 8.81ms, current 17.26ms)
```

Removing these from a 7.9MB log file removes about 3MB from the log file, and have not (so far) provided any value.

**To Reproduce**

This appears in the logs for certain debug zips. I'm not sure how this is enabled or if it's universal.

**Expected behavior**

This log surely does not need to be printed as much as it is. It should be more selective to print only when outlier latency jumps occur. The noise it generates creates problems with using the logs to troubleshoot issues, and if there is a latency jump that is an outlier that we should consider, we are more likely to miss it because of the spam.

**Environment:**
- v22.1.10

**Additional Context:**

I see that the codepath for this log line is `rpc/pkg/rpc/clock_offset.go:216`. Based on this I have labeled this with KV, but I'm not sure who owns this.

Please let me know if this needs more information, or reach out to me directly or here for any questions or details.

**Comment by erikgrinaker** [comment]
cc @knz

**Comment by knz** [comment]
Fwiw this logging is _already_ filtering to display only outliers. What you're seeing here is a cluster with abnormally irregular latencies. That, on its own, is worth investigating.

I also believe we have another issue on this topic which i will look up tomorrow.

**Comment by erikgrinaker** [comment]
Note that RPC heartbeats get head-of-line blocked by other RPC traffic, so this is not an accurate representation of the actual network latency. I've seen this spike as high as 6 seconds when the actual network latency was in the milliseconds.

**Comment by andrewbaptist** [comment]
Changing thresholds might remove a lot of these. Looking at a random customer log spanning about 1 day. There are >1M messages in the log of this type 
```
% grep "latency jump" */logs/crdb.ip* |  wc -l
 1084842
```

Changing instead to only look at the ones over 10ms reduces this a lot:
```
% grep "latency jump" */logs/crdb.ip* |  grep "current [1-9][0-9]\." | wc -l
   79360
```
They are still a little spammy but at least an order of magnitude less. but at the time of a problem (a slow node) it was more clear that they spiked upwards on that node at the time the node had a high CPU. 

A different filter looking at prev avg >5ms is even better:
```
% grep "latency jump" */logs/crdb.ip* |  grep "prev avg [5-9]" | wc -l
    3359
```

Once I figured out my bad node was node 151, I excluded it from the search and looked at the remaining messages.

```
% grep "latency jump" */logs/crdb.ip* | grep -v "^151" | grep "prev avg [5-9]" | wc -l
    2097
``` 

And looked at how many nodes were flagging this node as slow:
```
% grep "latency jump" */logs/crdb.ip* | grep -v "^151" | grep "prev avg [5-9]" | grep "rnode=151" | wc -l
    1702
```
 
So from all the other nodes, the majority of messages (1702 / 2097) = 81% were all pointing at the one bad node (151) and were all from the same 10-minute window. The remaining 395 were still a little spurious, but easy enough to ignore. 

One interesting (but somewhat funny) result of this is that nodes sometimes report themselves as having a latency jump. 
An example:
```
W221216 09:45:48.509302 31345166 2@rpc/pkg/rpc/clock_offset.go:216 ⋮ [n218,rnode=218,raddr=‹172.31.80.64:26256›,class=system,heartbeat] 1082714 latency jump (prev avg 0.56ms, current 1.02ms)
```
This is a good reason to change these thresholds.
